### PR TITLE
Address PR review comments for #8374

### PR DESCRIPTION
**Pull request by vtronkov**
**Description**:
Addressing PR comments from 8374

**Related issue(s)**:

Fixes https://github.com/hashgraph/hedera-services/issues/9244

**Notes for reviewer**:
About _"Replace uses of Pair with a dedicated record ScaledThrottle(DeterministicThrottle throttle, int logicalCapacityRequired) type. (Or similar.)"_ - I started implementing the change but it turns out that we are re-using the `ThrottleReqsManager` from the mono. In order to refactor that I will need to refactor mono code - `ThrottleReqsManager` and  `DeterministicThrottling` classes. So I'm not sure if it would be better if I create a tech debt story to refactor that after we retire the mono code. What do you think @tinker-michaelj ?

### CODE REVIEW

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/throttle/ThrottleAccumulator.java [hunk 1]
```diff
@@ -126,11 +130,25 @@ public boolean shouldThrottle(
...
-            @NonNull final HederaFunctionality queryFunction, @NonNull final Instant now, @NonNull final Query query) {
+            @NonNull final HederaFunctionality queryFunction,
+            @NonNull final Instant now,
+            @NonNull final Query query,
+            @Nullable final AccountID queryPayerId) {
         final var configuration = configProvider.getConfiguration();
+
+        if (queryPayerId == null) {
+            return false;
+        }
```
Comment (vtronkov) [inline hedera-node/hedera-app/src/main/java/com/hedera/node/app/throttle/ThrottleAccumulator.java:145]: Is this okay? If not how can I get the **payerId** in the `QueryWorkflowImpl.handleQuery` without calling `ingestChecker.runAllChecks(...)`?

Comment (netopyr) [reply hedera-node/hedera-app/src/main/java/com/hedera/node/app/throttle/ThrottleAccumulator.java:146]: `queryPayerId` is `null` if the query is free. There is a throttle defined for free queries. Thus, I think returning `false` is wrong. I guess we should only skip the `throttleExempt()`-check.

Comment (vtronkov) [reply hedera-node/hedera-app/src/main/java/com/hedera/node/app/throttle/ThrottleAccumulator.java]: Makes sense. Fixed.

**Review by mhess-swl — COMMENTED** [review]
LGTM, but someone who better understands throttles (e.g. @tinker-michaelj) should look at this

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/throttle/ThrottleAccumulator.java [hunk 2]
```diff
@@ -79,24 +78,29 @@
...
     private static final int UNKNOWN_NUM_IMPLICIT_CREATIONS = -1;
+
+    private EnumMap<HederaFunctionality, ThrottleReqsManager> functionReqs = new EnumMap<>(HederaFunctionality.class);
     private boolean lastTxnWasGasThrottled;
     private GasLimitDeterministicThrottle gasThrottle;
     private List<DeterministicThrottle> activeThrottles = Collections.emptyList();
 
+    private final ConfigProvider configProvider;
+    private final IntSupplier capacitySplitSource;
+    private final String throttleType;
```
Comment (netopyr) [inline hedera-node/hedera-app/src/main/java/com/hedera/node/app/throttle/ThrottleAccumulator.java:95]: I think it is better to use the enum here.

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/throttle/ThrottleAccumulator.java [hunk 3]
```diff
@@ -531,7 +542,8 @@ public void applyGasConfig() {
         gasThrottle = new GasLimitDeterministicThrottle(capacity);
 
         log.info(
-                "Resolved consensus gas throttle -\n {} gas/sec (throttling {})",
+                "Resolved {} gas throttle -\n {} gas/sec (throttling {})",
+                throttleType,
```
Comment (netopyr) [inline hedera-node/hedera-app/src/main/java/com/hedera/node/app/throttle/ThrottleAccumulator.java:546]: Is this the only place we use the `ThrottleType.type()`? If so, I think we can remove it and use the enum's name directly. It is just us who are reading the logs.

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/Hedera.java [hunk 4]
```diff
@@ -758,9 +760,9 @@ private void restart(
         logger.info("Initializing ThrottleManager");
         this.throttleManager = new ThrottleManager();
 
-        this.backendThrottle = new ThrottleAccumulator(() -> 1, configProvider);
-        this.frontendThrottle =
-                new ThrottleAccumulator(() -> platform.getAddressBook().getSize(), configProvider);
+        this.backendThrottle = new ThrottleAccumulator(() -> 1, configProvider, BACKEND_THROTTLE.type());
```
Comment (netopyr) [inline hedera-node/hedera-app/src/main/java/com/hedera/node/app/Hedera.java:763]: `() -> 1` can be `static`

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/throttle/ThrottleAccumulator.java [hunk 5]
```diff
@@ -543,7 +555,8 @@ private ThrottleGroup<HederaFunctionality> hapiGroupFromPbj(
     }
 
     private void logResolvedDefinitions(final int capacitySplit) {
-        var sb = new StringBuilder("Resolved handle throttles (after splitting capacity ")
+        var sb = new StringBuilder("Resolved " + throttleType + " ")
```
Comment (netopyr) [inline hedera-node/hedera-app/src/main/java/com/hedera/node/app/throttle/ThrottleAccumulator.java:558]: Concatenating a String and passing the result to a StringBuilder is inefficient. It is better to use `append()` already here.

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/throttle/ThrottleAccumulator.java (hunk 3 above)
Comment (vtronkov) [reply hedera-node/hedera-app/src/main/java/com/hedera/node/app/throttle/ThrottleAccumulator.java]: We use it for another log as well but it makes sense to use the enum's name, yes 👍

**Review by tinker-michaelj — APPROVED** [review]
LGTM!

**Comment by tinker-michaelj** [comment]
>In order to refactor that I will need to refactor mono code - ThrottleReqsManager and DeterministicThrottling classes. So I'm not sure if it would be better if I create a tech debt story to refactor that after we retire the mono code. What do you think @tinker-michaelj ?

Ah I missed that, I think we can leave as written here, ty @vtronkov !

**Review by mhess-swl — APPROVED** [review]

**Pull request merged**